platform/ios/src/MGLAPIClient.m

```diff
@@ -88,7 +88,7 @@ - (NSURLRequest *)requestForEvents:(NS_ARRAY_OF(MGLMapboxEventAttributes *) *)ev
 
 - (void)setupBaseURL {
     NSString *testServerURL = [[NSUserDefaults standardUserDefaults] stringForKey:@"MGLTelemetryTestServerURL"];
-    if (testServerURL) {
+    if (testServerURL && [testServerURL hasPrefix:@"https://"]) {
```

Comment:
I went whole hog and refactored everything to use NSURL in 2510d116bd88dc5a7c5e46665d4b63a63d9cc0ee.